src/main/java/gameframework/motion/MoveStrategyRandom.java

```diff
@@ -28,6 +28,8 @@ public SpeedVector getSpeedVector() {
 		case 3:
 			currentMove.setDirection(new Point(0, 1));
 			break;
+		default:
+			return currentMove;
```

Comment:
please throw a runtime exception here to tell the reader that this will never happen.